### PR TITLE
fix: correct Notification interface (#7073)

### PR DESCRIPTION
**Pull request by marswong**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/notification/__tests__/index.test.js [hunk 1]
```diff
@@ -51,17 +51,27 @@ describe('notification', () => {
...
         description: 'This is the content of the notification.',
       });
-      expect(document.querySelectorAll(`${iconPrefix}-${type}`).length).toBe(1);
+      expect(document.querySelectorAll(`${iconPrefix}-${icon}`).length).toBe(1);
     };
-    ['success', 'info', 'warning', 'error'].forEach((type) => {
-      openNotificationWithIcon(type);
+    ['success', 'info', 'warning', 'error'].forEach((icon) => {
+      openNotificationWithIcon(icon);
     });
```
Comment (afc163) [inline components/notification/__tests__/index.test.js]: 此处参数名还是保持 `type`，比较好理解。

File: components/notification/index.tsx [hunk 2]
```diff
@@ -203,10 +203,10 @@ const api = {
   },
 };
 
-['success', 'info', 'warning', 'error'].forEach((type) => {
-  api[type] = (args: ArgsProps) => api.open({
+['success', 'info', 'warning', 'error'].forEach((icon) => {
+  api[icon] = (args: ArgsProps) => api.open({
     ...args,
-    type,
+    icon,
```
Comment (afc163) [inline components/notification/index.tsx]: 同上。

```
icon: type,
```

**Comment by afc163** [comment]
感觉复用一个 API 还是不妥，如果想用 `icon: 'info'` 这个图标，就挂了。

**Comment by marswong** [comment]
确实。但是如果 `type` 是硬编码的话貌似也不妥🙈，我再想想

**Comment by afc163** [comment]
试试内部和外部接口分开。

**Review by afc163 — APPROVED** [review]

File: components/notification/index.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ export interface ArgsProps {
   placement?: notificationPlacement;
   style?: string;
   className?: string;
-  type?: string;
+  readonly type?: string;
```
Comment (afc163) [inline components/notification/index.tsx]: 👍 学习了

**Comment by afc163** [comment]
ref https://github.com/ant-design/ant-design/issues/7073

**Pull request merged**